src/der.rs

```diff
@@ -77,6 +77,20 @@ pub(crate) fn expect_tag_and_get_value_limited<'a>(
     Ok(inner)
 }
 
+pub(crate) fn nested_limited<'a, F, R, E: Copy>(
+    input: &mut untrusted::Reader<'a>,
+    tag: Tag,
+    error: E,
+    decoder: F,
+    size_limit: usize,
+) -> Result<R, E>
+where
+    F: FnOnce(&mut untrusted::Reader<'a>) -> Result<R, E>,
```

Comment:
Recommend using `impl FnOnce(..) -> ..` for the `decoder` argument directly instead to make this more concise and avoid the type argument indirection.

Reply:
Done. Thanks